llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -32,18 +32,19 @@ using namespace llvm;
 namespace {
 // A list of supported preprocessing directives with their
 // internal token kinds and names.
-struct {
+struct PreprocessorDir {
   tgtok::TokKind Kind;
-  const char *Word;
-} PreprocessorDirs[] = {
-  { tgtok::Ifdef, "ifdef" },
-  { tgtok::Ifndef, "ifndef" },
-  { tgtok::Else, "else" },
-  { tgtok::Endif, "endif" },
-  { tgtok::Define, "define" }
+  StringRef Word;
 };
 } // end anonymous namespace
 
+static constexpr PreprocessorDir PreprocessorDirs[] = {
```

Comment:
Just `constexpr`. This has internal linkage even without `static`

Reply:
Done.